apps/dgst.c

```diff
@@ -467,9 +489,17 @@ int do_fp(BIO *out, unsigned char *buf, BIO *bp, int sep, int binout,
     if (binout) {
         BIO_write(out, buf, len);
     } else if (sep == 2) {
+        int backslash = 0;
+        file = newline_escape_filename(file, &backslash);
+
         for (i = 0; i < (int)len; i++)
-            BIO_printf(out, "%02x", buf[i]);
+            if (i == 0 && backslash == 1)
+                BIO_printf(out, "\\%02x", buf[i]);
+            else
+                BIO_printf(out, "%02x", buf[i]);
+
         BIO_printf(out, " *%s\n", file);
+        OPENSSL_free((char *)file);
```

Comment:
Might it be worth inlining the string print code and avoiding the allocation/free and copy?

Reply:
I took the function approach in case there was ever a need to reuse it later on.  The `newline_escape_filename` function could be ported out to apps.c